module/VuFindSearch/src/VuFindSearch/ParamBag.php

```diff
@@ -162,7 +162,7 @@ public function add($name, $value)
             $this->params[$name] = [];
         }
         if (is_array($value)) {
-            $this->params[$name] = array_merge($this->params[$name], $value);
+            $this->params[$name] = array_unique(array_merge($this->params[$name], $value));
```

Comment:
I think it probably makes sense to apply the array_unique later in the function, since as coded here, it's only going to get applied if `$value` is an array, but a single duplicate value could also potentially be passed in.

I also wonder if we need a third `$deduplicate` parameter here (perhaps with a default value of true), so the deduplication could be explicitly bypassed in situations where a duplicate value is needed. (I can't immediately think of such a situation, but I imagine it's possible there could be a reason to repeat a value in some parameter somewhere).

Also interested in whether @EreMaijala has any thoughts/opinions on this.

Reply:
As discussed in the community call, this looks fine to me, but the single value case should be handled as well.

Reply:
Thanks for the feedback! It has been quite easy to implement the changes, please have another look!